src/Domain/ValueObjects/ShoppingItemFilterEnum.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Lindyhopchris\ShoppingList\Domain\ValueObjects;
+
+class ShoppingItemFilterEnum
+{
+    /**
+     * @var string
+     */
+    private string $all;
+
+    /**
+     * @param string $all
+     */
+    public function __construct(string $all)
+    {
+        $this->all = $all;
```

Comment:
Required: at the moment, your class accepts _any_ string, whereas it should only accept "valid" strings, i.e. one of the three strings that represent the different types of filters. In the constructor, you need to check that the value you've been given is one of those 3 valid strings, and if not throw a `InvalidArgumentException`.